wallet/src/hooks/User/index.jsx

```diff
@@ -165,11 +167,23 @@ export const UserProvider = ({ children }) => {
           emptyStoreTimber();
           Storage.shieldAddressSet();
         } else {
-          setLastBlock(parsed.data.block);
-          await blockProposedEventHandler(parsed.data, [zkpPrivateKey], [nullifierKey]);
+          setLastBlock(parsedEvent.data.block);
+          await blockProposedEventHandler(parsedEvent.data, [zkpPrivateKey], [nullifierKey]);
         }
       }
-      // TODO Rollback Handler
+
+      // if (parsedEvent.type === 'rollback') {
```

Comment:
Nah, I just kept it here, because this is exactly the next thing I am planning to work on :) Will be removed, or more like converted into real code in the next MR if thats OK with you :)